src/Diagnostics.RuntimeHost/Services/HealthCheckService/HealthCheckService.cs

```diff
@@ -54,6 +58,14 @@ public async Task RunHealthCheck()
             await RetryHelper.RetryAsync(HealthCheckPing, "Healthping", "", 3, 100);
         }
 
+        public async Task<IEnumerable<HealthCheckResult>> RunDependencyCheck(DataProviders.DataProviders dataProviders)
+        {
+            var results = new List<HealthCheckResult>();
+            DiagnosticDataProvider dp = ((LogDecoratorBase)dataProviders.Kusto).DataProvider;
```

Comment:
should we get all data providers here and run health check for each one of them?

Reply:
I have that change and it will go into a different PR. This was just part 1 of something that works and I wanted to get it reviewed rather than send one big PR